### ISSUE TITLE
MethodReader should recycle objects for multiple argument calls.

### ISSUE THREAD

**Issue by peter-lawrey** [OpenHFT/Chronicle-Wire#56]

**Comment by peter-lawrey** [comment]
I have updated the bom as well.

**Comment by RobAustin** [comment]
Thanks Peter, I’ll release it now

**Comment by RobAustin** [comment]
looks like this may have caused an issue

```
written to files ok
2017-10-23 12:47:06.888 [main/runnerCfg-4] WARN  net.openhft.affinity.LockInventory - No reservable CPU for Thread[main/runnerCfg-4,5,main]
2017-10-23 12:47:06.902 [main/runnerCfg-6] WARN  c.r.f.dto.status.HeartbeatDefinition - creating default HeartbeatDefinition for pricer2
2017-10-23 12:47:06.927 [main/runnerCfg-6] WARN  net.openhft.affinity.LockInventory - No reservable CPU for Thread[main/runnerCfg-6,5,main]
Waiting for output to start from config-out took 150ms
Waiting for output to  stop from config-out took 150ms
Waiting for output to start from config-out took 150ms
Waiting for output to  stop from config-out took 150ms
2017-10-23 12:47:09.262 [main] WARN  n.o.chronicle.wire.MethodReader - Unable to support overloaded methods, ignoring one of closeQuietly
2017-10-23 12:47:09.318 [main] WARN  n.o.chronicle.wire.MethodReader - Unable to support overloaded methods, ignoring one of closeQuietly
2017-10-23 12:47:09.325 [main/runnerCfg-6] WARN  com.rbccm.fx.multipricer.MultiPricer - Failure to dispatch message: ldr [java.lang.Object@7e84b2fa, null]
java.lang.ClassCastException: java.lang.Object cannot be cast to java.lang.String
    at net.openhft.chronicle.wire.ScalarStrategy.readUsing(ScalarStrategy.java:75) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.Wires.object0(Wires.java:475) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.BinaryWire$BinaryValueIn.object(BinaryWire.java:2021) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.MethodReader.lambda$addParseletForMethod$7(MethodReader.java:254) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.BinaryWire$BinaryValueIn.sequence(BinaryWire.java:2797) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.MethodReader.lambda$addParseletForMethod$8(MethodReader.java:270) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.WireParser.parseOne(WireParser.java:41) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.WireParser.accept(WireParser.java:49) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.MethodReader.readOne(MethodReader.java:312) ~[chronicle-wire-1.9.21.jar:na]
    at com.rbccm.fx.run.Runner.runLoop(Runner.java:648) ~[api-1.8.19-SNAPSHOT.jar:na]
    at com.rbccm.fx.run.Runner.run(Runner.java:479) ~[api-1.8.19-SNAPSHOT.jar:na]
    at com.rbccm.fx.run.Runner.run(Runner.java:268) ~[api-1.8.19-SNAPSHOT.jar:na]
    at com.rbccm.fx.run.ServiceCfg.run(ServiceCfg.java:190) ~[api-1.8.19-SNAPSHOT.jar:na]
    at com.rbccm.fx.run.AllMultiTester.lambda$null$2(AllMultiTester.java:118) ~[test-classes/:na]
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_141]
    at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_141]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_141]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_141]
    at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_141]
2017-10-23 12:47:09.372 [main] WARN  n.o.chronicle.wire.MethodReader - Unable to support overloaded methods, ignoring one of closeQuietly
2017-10-23 12:47:09.379 [main/runnerCfg-6] WARN  com.rbccm.fx.multipricer.MultiPricer - Failure to dispatch message: ldr [java.lang.Object@7e84b2fa, null]
java.lang.ClassCastException: java.lang.Object cannot be cast to java.lang.String
    at net.openhft.chronicle.wire.ScalarStrategy.readUsing(ScalarStrategy.java:75) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.Wires.object0(Wires.java:475) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.BinaryWire$BinaryValueIn.object(BinaryWire.java:2021) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.MethodReader.lambda$addParseletForMethod$7(MethodReader.java:254) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.BinaryWire$BinaryValueIn.sequence(BinaryWire.java:2797) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.MethodReader.lambda$addParseletForMethod$8(MethodReader.java:270) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.WireParser.parseOne(WireParser.java:41) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.WireParser.accept(WireParser.java:49) ~[chronicle-wire-1.9.21.jar:na]
    at net.openhft.chronicle.wire.MethodReader.readOne(MethodReader.java:312) ~[chronicle-wire-1.9.21.jar:na]
    at com.rbccm.fx.run.Runner.runLoop(Runner.java:648) ~[api-1.8.19-SNAPSHOT.jar:na]
    at com.rbccm.fx.run.Runner.run(Runner.java:479) ~[api-1.8.19-SNAPSHOT.jar:na]
    at com.rbccm.fx.run.Runner.run(Runner.java:268) ~[api-1.8.19-SNAPSHOT.jar:na]
    at com.rbccm.fx.run.ServiceCfg.run(ServiceCfg.java:190) ~[api-1.8.19-SNAPSHOT.jar:na]
    at com.rbccm.fx.run.AllMultiTester.lambda$null$2(AllMultiTester.java:118) ~[test-classes/:na]
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_141]
    at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_141]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_141]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_141]
    at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_141]
```

**Comment by peter-lawrey** [comment]
This should be fixed now.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
